packages/jest-resolve/src/__tests__/resolve.test.js

```diff
@@ -78,6 +78,34 @@ describe('findNodeModule', () => {
       paths: (nodePaths || []).concat(['/something']),
     });
   });
+
+  it('removes query strings from path before resolving', () => {
+    const nodePaths = process.env.NODE_PATH
+      ? process.env.NODE_PATH
+          .split(path.delimiter)
+          .filter(Boolean)
+          .map(p => path.resolve(resolvedCwd, p))
+      : null;
+
+    const newPath = Resolver.findNodeModule('./test.jpg?sizes=300w', {
```

Comment:
Might be worthwhile to use `test.jpg?sizes=300w+600w&foo=bar&baz` to encompass a complex example.